### PR TITLE
daemonize 1.7.7

### PR DESCRIPTION
**Pull request by matteocng**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Does your submission pass `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Have you built your formula locally prior to submission with `brew install <formula>`?

### CODE REVIEW

File: Formula/daemonize.rb [hunk 1]
```diff
@@ -18,4 +18,8 @@ def install
     system "make"
     system "make", "install"
   end
+
+  test do
+    system "#{bin}/daemonize"
```
Comment (ilovezfs) [inline Formula/daemonize.rb]: Can we have this test do something real? Or at minimum assert match the version?

**Comment by matteocng** [comment]
In case someone else bumps into a Mavericks test-bot build failing the test with "can't convert String into Integer": Ruby accepts a symbolic (String) chmod  mode argument [since 1.9.3-preview1](/ruby/ruby/commit/0a1e7d), but the Mavericks executor is running 1.8.7-p374_1. The error was fixed by changing `chmod "+x"` to `chmod "0700"`.

File: Formula/daemonize.rb [hunk 2]
```diff
@@ -18,4 +18,23 @@ def install
...
+    system "#{sbin}/daemonize", "-p", pid_file, dummy_script_file
+    assert((File.exist? pid_file), "The file containing the PID of the child process was not created.")
+    sleep_interval = 0.3
+    sleep_limit = 4
+    slept_total = 0
+    # sleep while waiting for the dummy script to finish
+    while File.exist?(output_file) == false && slept_total < sleep_limit
+      sleep(sleep_interval)
+      slept_total += sleep_interval
+    end
```
Comment (bfontaine) [inline Formula/daemonize.rb]: Please use `sleep 4` directly instead of a loop.

File: Formula/daemonize.rb [hunk 3]
```diff
@@ -18,4 +18,23 @@ def install
     system "make"
     system "make", "install"
   end
+
+  test do
+    dummy_script_file = testpath/"script.sh"
+    output_file = testpath/"outputfile.txt"
+    pid_file = testpath/"pidfile.txt"
+    dummy_script_file.write format("echo version %s >> %s", version.to_s, output_file)
```
Comment (bfontaine) [inline Formula/daemonize.rb]: Please use string interpolation with `#{}` instead of `format`.

File: Formula/daemonize.rb [hunk 4]
```diff
@@ -18,4 +18,17 @@ def install
     system "make"
     system "make", "install"
   end
+
+  test do
+    dummy_script_file = testpath/"script.sh"
+    output_file = testpath/"outputfile.txt"
+    pid_file = testpath/"pidfile.txt"
+    dummy_script_file.write "echo #{version} >> #{output_file}"
```
Comment (UniqMartin) [inline Formula/daemonize.rb]: This happens to work on OS X, but relies on some ancient compatibility feature. It's preferable to add a shebang line to the shell script to make it clear to the operating system how to execute the script. You can create a multi-line file like so (it's also a good idea to quote strings in the shell script):

```ruby
dummy_script_file.write <<-EOS.undent
  #!/bin/sh
  echo "#{version}" >> "#{output_file}"
EOS
```

Other than that the code looks good. Many thanks for adding a test!

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/283e53d065b7d9af4029e771aaf18628a6f8b1b2. Thank you for your contribution to Homebrew @matteocng; we appreciate it! :smiley_cat:

**Pull request closed without merging**